uPortal-url/src/main/java/org/apereo/portal/url/IAuthUrlCustomizer.java

```diff
@@ -0,0 +1,22 @@
+package org.apereo.portal.url;
+
+import javax.servlet.http.HttpServletRequest;
+
+/** Customize CAS or any other authentication URL. */
+public interface IAuthUrlCustomizer {
+    /**
+     * Does this customizer supports the current HTTP request.
+     *
+     * @param request
+     * @return
+     */
+    boolean supports(HttpServletRequest request, String url);
+
+    /**
+     * Customize the supplied external login URL.
+     *
+     * @param url
```

Comment:
`url` but no `request` param?

Reply:
The 2 methods have the sames params, but each one had a different param :laughing: 
=> Solved

for notes I only moved the working code from my old previous PR, applied it on my portal instance (without a lot of change) and finally come back to push it, so sorry i didn't take the time to look more on the code.